Src/DependencyCollector/Shared/ServiceBusDiagnosticsListener.cs

```diff
@@ -0,0 +1,171 @@
+﻿namespace Microsoft.ApplicationInsights.DependencyCollector.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Threading.Tasks;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation;
+
+    /// <summary>
+    /// Implements ServiceBus DiagnosticSource events handling.
+    /// </summary>
+    internal class ServiceBusDiagnosticsEventHandler
+    {
+        public const string DiagnosticSourceName = "Microsoft.Azure.ServiceBus";
+        private const string StatusPropertyName = "Status";
+        private const string EntityPropertyName = "Entity";
+        private const string EndpointPropertyName = "Endpoint";
+        private const string ExceptionPropertyName = "Exception";
+
+        // We want to reflect in the dependency type that it's one of the queue operations.
+        // This way, UI could have special representation for all kinds of queues
+        // Not knowing particular service
+        private const string DependencyType = "Queue.ServiceBus";
+
+        private readonly TelemetryClient telemetryClient;
+
+        // Every fetcher is unique for event payload and particular property into this payload
+        // when we first receive an event and require particular property, we cache fetcher for it
+        // There are just a few (~10) events we can receive and each will a have a few payload fetchers
+        // I.e. this dictionary is quite small, does not grow up after service warm-up and does not require clean up
+        private readonly Dictionary<string, Dictionary<string, PropertyFetcher>> propertyFetchers =
```

Comment:
Instead of a "dictionary of dictionaries" consider a single Dictionary keyed on a ValueTuple of the two strings.
(If you can't use ValueTuple, consider writing a struct with two strings and a custom GetHashCode - don't just concat the two strings because the idea is to avoid heap allocations for lookup).

Reply:
right